### ISSUE TITLE
zstd.ZstdError: zstd decompress error: Unknown frame descriptor

### ISSUE THREAD

**Issue by thestick613** [indygreg/python-zstandard#79]
```
files = glob.glob('/home/work/frequency_analyzer/remote_mount/*.zstd')
for file_name in files:
    with open(file_name, "rb") as fh:
        dctx = zstd.ZstdDecompressor()
        reader = dctx.stream_reader(fh)
        while True:
            chunk = reader.read(65536)
            if not chunk:
                break

Traceback (most recent call last):
  File "load_frequency.py", line 120, in <module>
    chunk = reader.read(65536)
zstd.ZstdError: zstd decompress error: Unknown frame descriptor
```

The exception was raised towards the end of a very large file (200GB). What could be the cause of it? Is the file damaged? If so, can i skip through the faulty data, or is the whole operation compromised.

**Comment by bzupnick** [comment]
I'm having a similar problem. Has this been solved or understood better by anyone?

**Comment by 1e100** [comment]
I had this issue and it turned out that the file just wasn’t compressed at all, it merely got caught with a glob, but was plain text. Try decompressing with command line zstd

**Comment by indygreg** [comment]
`Unknown frame descriptor` means the incoming data doesn't have a zstd frame header. This either means the data isn't zstd compressed or was written in _magicless_ mode and the decoder didn't also engage _magicless_ mode.

In previous versions of this library, it was possible for some bytes to get _stuck_ in output buffers due to how `flush()` and `close()` were handled. But this should be fixed as of the 0.15 release. If you are using the APIs as documented in examples, things should _just work_.

Since this issue doesn't appear to be actionable, I'm closing it.
